src/server/pkg/serviceenv/config.go

```diff
@@ -75,7 +75,8 @@ type StorageConfiguration struct {
 	StoragePutFileConcurrencyLimit int    `env:"STORAGE_PUT_FILE_CONCURRENCY_LIMIT,default=100"`
 	StorageGCPolling               string `env:"STORAGE_GC_POLLING"`
 	StorageGCTimeout               string `env:"STORAGE_GC_TIMEOUT"`
-	StorageCompactionMaxFanIn      int    `env:"STORAGE_COMPACTION_MAX_FANIN"`
+	StorageCompactionMaxFanIn      int    `env:"STORAGE_COMPACTION_MAX_FANIN,default=50"`
+	StorageMaxOpenFileSets         int    `env:"STORAGE_MAX_OPEN_FILESETS,default=50"`
```

Comment:
We need to be more consistent with the naming convention for these configurations. I think the namespacing approach makes sense (`STORAGE_GC_*`, `STORAGE_COMPACTION_*`, etc). We probably want this to apply to filesets as well, so it would be `STORAGE_FILESETS_*` I guess. What do you think?

Reply:
`STORAGE_FILESETS_MAX` ?  There are also possible configurations that don't really make sense. Like fan-in=5, open-filesets=4.  Would that deadlock or just compact 4 at a time?

Reply:
I was thinking `STORAGE_FILESETS_MAX_OPEN`, but I have a feeling @jdoliner might not like that naming convention. I think it could be useful to have things namespaced, but it reads a bit weird. Also, for your example, those configurations are orthogonal. Fan-in applies to open readers, while open filesets are the in-memory filesets that are used for buffering writes. It might make sense to capture this difference better in the configuration names.

Reply:
I think I like the original name `STORAGE_MAX_OPEN_FILESETS` reads the best to me, but I don't have a strong opinion on it. I don't think users are ever going to be able to read an env-var and understand what it's for without documentation. So as long as the names aren't totally meaningless (which none of the proposed ones are) I think they'll work approximately the same in terms of user experience. That means you guys should probably just choose the one you feel organizes the env vars the best with respect to the code and how it's implemented.

Reply:
Okay, I prefer the namespaced approach even though it does not read best. I think it will be easier to work with and organize with this naming convention.